export.md

```diff
@@ -1,4 +1,14 @@
 # FHIR Bulk Data Export API Proposal
+ 
+## Security Considerations
+
+When initiating any request with the FHIR resource server, the client first MUST use [Transport Layer Security (TLS) Protocol Version 1.2 (RFC5246)](https://tools.ietf.org/html/rfc5246) to authenticate the identity of the FHIR server and to establish an encrypted, integrity-protected link for securing all exchanges between the client and the server involved in the exchange.  All exchanges described herein between a client and a server MUST be secured using TLS V1.2.  Use of mutual TLS is OPTIONAL.  
+
+Prior to kicking off a bulk-data request, the authorization server responsible for authorizing access to the desired FHIR resources MUST have authenticated the client and issued it one or more access tokens for use in requesting the export of FHIR resources, status information, and file downloads (which could involve multiple servers).  For this purpose, the OAuth 2.0 based SMART Backend Services: Authorization Guide is RECOMMENDED.
+
+The decision whether to issue a single access token for use for all three purposes, or to issue multiple tokens is a risk-management decision to be made by the organization responsible for managing access to the resources. To minimize risks associated with token redirection, the scope of each access token SHOULD encompass, and be limited to, the resources requested.  Also, in some cases, the server that receives status queries may be different from the FHIR resource server.  So the use of separate access tokens for requesting FHIR resources and for requesting status information is RECOMMENDED. The scope of resources applicable to the access token required (optionally) for file download will be identical to (or a subset of) the token used for the initial request.  So the authorization server MAY choose to allow the use of the same token for both purposes, or it MAY choose to issue a new token at the time needed for file download.
+
+Regardless of how access tokens are obtained and how many tokens are issued, the requests outlined below MUST include an ```Authorization``` header containing a bearer token received from the responsible authorization server.   Each access token MUST include the identity of each intended recipient (audience).  Upon receipt of a request, the server to whom the token is presented MUST confirm that the token was issued by the applicable authorization server, that the token is intended for that server’s use, that the scope includes the resources requested, and that the time of use is within the valid use period.   If not, the server SHOULD return an invalid_client response.    
```

Comment:
Two quick issues here @bakerdb 

1. If the export API isn't being used with SMART backend services or another OAuth based security spec, this token shouldn't be present. So this kind of detail to me belongs in the security spec rather than the data export API spec, since it could vary.

2. The "must include" language is a bit tricky, since technically the access token is opaque to the client, and it may not "include' anything if it's just a database id or correlation handler.

I think it'd be cleaner to avoid this language altogether in export.md; we already say that SMART Backend Services is "recommended", which to me is the right touchpoint between the specs.

Reply:
Thanks -- for (2) I agree your language is consistent with RFC 6750. The RFC language is used to describe expectations of the authz server ("it is important for the authorization server to include ... [stuff]... in the token.") whereas here the obligations are bouncing back and forth a bit between server and client. But anyway, it's a detail and I don't mind deeply :-)

For (1), I think it will help to discuss in a call. While access is asynchronous and completing an export may take a long time, any given authz token only has a short lifetime, which is the case regardless of how we scope the audience.

Reply:
I see your point re #1.  I do think the topic needs to be addressed, as the situation in which multiple endpoints are involved is likely to arise.   I agree we should discuss this.

Reply:
@jmandel I agree that a discussion regarding when to require new tokens is best placed in the Authorization Guide, rather than the Bulk Access Implementation Guide.  I will generate a new PR accordingly.